### DISCUSSION TITLE
Profile directory change - community discussion.

### DISCUSSION THREAD

**Discussion by Alex313031** [Alex313031/thorium#3] [Ideas]
This is for anyone who wants to answer.
I am considering changing the default profile dir for Thorium from $HOME/.config/chromium to $HOME/.config/thorium.
This would allow running chromium simultaneously, and make sure that someone migrating from chromium to thorium wouldn't have user profile errors from version differences.
@pierro78

**Comment by pierro78** [comment]
good idea !

**Comment by Alex313031** [comment]
@pierro78 I've found that generally you cant upgrade or downgrade more than 3 to 4 major version differences, without user profile errors. Instead of doing it the "proper" way, which I simply don't know how to do and couldn't find info on where in the source code you specify profile dirs, I will instead be adding --user-data-dir=~/.config/thorium to the .desktop file and /usr/bin file.
Also, the .desktop file has had a desktop action to open content_shell, but it didn't work until I figured out .desktop files cant call binaries from /opt, and so I modified it to point to the wrapper I made for content_shell in /usr/bin/ that is just modeled after debians wrapper in their package chromium-shell. It now works as intended, so you can launch the content shell with either the Thorium-Shell.desktop file, or right clicking the Thorium launcher in your desktop environment and their should be an option to Open Content Shell after the New Window and New Incognito Window options.
I am also adding a bug to BUGS.md; chromium as of 97 when one selects it as the default browser, it properly sets itself as so, but on next start it still thinks it isn't the default and so pops up the annoying "set as default browser?" infobar. I will be disabling this by also setting --no-default-browser-check in the .desktop file. Anyone who can install thorium from the .deb should be able to set it as their default browser without having an annoying infobar directing them to do so. I will still be testing each release without this flag to see when (if?) it gets resolved. It doesn't appear to be DE or WM dependent, as it happens on my dev machine (openbox) as well as my tests in virtual machines on ubuntu, popos. (gnome 4 and gnome 3).

Also, another question I'd like your opinion on. As you probably know, the package includes three extra stuff, namely the chromium-sandbox, needed on older distros like ubuntu 16.04 and debian 9, the content_shell, and chromedriver. However, right now it installs two copies of the chromedriver executable, one in /opt/chromium.org/thorium and one in /usr/bin, but this adds another 50mb to the package. I am considering removing it from /opt/chromium.org/thorium and just putting it in /usr/bin (the way the debian chromium-driver package does) Do you think this is a good idea. The only use case I can see this hurting is that a dev might want to directly call the one in /opt to use it vanilla style, while modifying the /usr/bin one to run it headlessly, or append cmdline flags. But if someone is savvy enough to be using chromedriver in these ways, then they most likely know how to just make a bash script to do this for them.

One last thing. I will also be adding a desktop action called Open in Safe Mode, which will have the --no-experiments flag to disable all chrome://flags. This allows one to diagnose/disable a flag thats causing issues without resetting all the flags (this has been useful for me because I regularly have 100+ of the flags in chrome://flags set and its annoying to have to go back and read and redo all the ones I want.
Also considering adding a desktop action that opens thorium in debug mode, which is insecure and slower, but allows native inspection. You cant do any heavy debugging though, because part of my optimizations for thorium are excluding debug symbols, enabling stripping, and disabling full stack dumps as these add bloat and slight overhead when running. Interestingly, Google Chrome has the first two set, but still has full stack dumps enabled.

**Comment by Alex313031** [comment]
@pierro78 Keep in mind that once the profile folder change has landed, you should sudo apt purge thorium, then BEFORE INSTALLING the new release, create the thorium dir with mkdir ~/.config/thorium and just copy everything in ~/.config/chromium to that dir. Then you can install the new release and everything should work. On another note though, this also means you could test, say @RobRich999 releases and run them side by side independent of each other.

Another thing. I disabled FLOC in Thorium as I texted you about, but if you are still using the same profile as a release that still had FLOC, it will still be enabled. To disable it go to the chromium or thorium profile dir and delete the 'Floc' folder.

**Comment by Alex313031** [comment]
@pierro78 @midzer @gz83 Marking as completed as the user profile dir has been changed in source code for all platforms as of two weeks ago.
